internal/getproviders/registry_client.go

```diff
@@ -181,6 +196,8 @@ func (c *registryClient) PackageMeta(ctx context.Context, provider addrs.Provide
 		target.OS,
 		target.Arch,
 	))
+	ctx, span := tracing.Tracer().Start(ctx, "opentofu.provider.meta")
```

Comment:
`suggestion`
Maybe moving this after the definition of `endpointURL` and add that as attribute? 🤔